src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -30,6 +30,8 @@ protected MessageBody(Frame context)
 
         public bool RequestUpgrade { get; protected set; }
 
+        public virtual bool IsEmpty => false;
+
         public Task<int> ReadAsync(ArraySegment<byte> buffer, CancellationToken cancellationToken = default(CancellationToken))
         {
             var task = PeekAsync(cancellationToken);
```

Comment:
Not necessary as part of this PR, but I agree with @pakrym that PeekAsync can be changed back to ReadAsync now that CopyToAsync doesn't use it.